model/events/LtiAgsListener.php

```diff
@@ -108,60 +89,49 @@ public function onDeliveryExecutionResultsRecalculated(DeliveryExecutionResultsR
-            $scoreTotalMax = null;
-
-            foreach ($session->getAllVariables()->getArrayCopy() as $variable) {
-                if ($variable instanceof OutcomeVariable) {
-                    $value = $variable->getValue();
-
-                    if (!$value instanceof QtiScalar) {
-                        continue;
-                    }
-
-                    if ($variable->getIdentifier() === 'SCORE_TOTAL') {
-                        $scoreTotal = $value->getValue();
-                    }
+        $launchData = $this->getLtiContextRepository()->findByDeliveryExecutionId($event->getDeliveryExecutionId());
+
+        $scoreTotal = null;
+        $scoreTotalMax = null;
+        $scoreTotalMicrotime = null;
+        foreach ($event->getVariables() as $variable) {
+            $variable = array_pop($variable)->variable;
+
+            if ($variable instanceof OutcomeVariable) {
+                if ($variable->getIdentifier() === 'SCORE_TOTAL') {
+                    $scoreTotal = $variable->getValue();
+                    $scoreTotalMicrotime = $variable->getEpoch();
+                }
 
-                    if ($variable->getIdentifier() === 'SCORE_TOTAL_MAX') {
-                        $scoreTotalMax = $value->getValue();
-                    }
+                if ($variable->getIdentifier() === 'SCORE_TOTAL_MAX') {
+                    $scoreTotalMax = $variable->getValue();
+                }
 
-                    if ($scoreTotal !== null && $scoreTotalMax !== null) {
-                        break;
-                    }
+                if ($scoreTotal !== null && $scoreTotalMax !== null) {
+                    break;
                 }
             }
-
-            $this->queueSendAgsScoreTaskWithScores(
-                'AGS score send on test finish',
-                $user->getLaunchData(),
-                $deliveryExecution,
-                $scoreTotal,
-                $scoreTotalMax,
-                $this->isManualScored($session)
-                    ? ScoreInterface::GRADING_PROGRESS_STATUS_PENDING_MANUAL
-                    : ScoreInterface::GRADING_PROGRESS_STATUS_FULLY_GRADED,
-                DateHelper::formatMicrotime($deliveryExecution->getFinishTime())
-            );
         }
+
+        $this->queueSendAgsScoreTaskWithScores(
+            'AGS score send on test finish',
+            $launchData,
+            $event->getDeliveryExecutionId(),
+            $scoreTotal,
+            $scoreTotalMax,
+            $event->getIsManualScored()
+                ? ScoreInterface::GRADING_PROGRESS_STATUS_PENDING_MANUAL
+                : ScoreInterface::GRADING_PROGRESS_STATUS_FULLY_GRADED,
+            DateHelper::formatMicrotime($scoreTotalMicrotime)
+        );
     }
 
     private function queueSendAgsScoreTaskWithScores(
         string $taskLabel,
         LtiLaunchData $ltiLaunchData,
-        DeliveryExecutionInterface $deliveryExecution,
+        string $deliveryExecutionId,
```

Comment:
Its because the new event don't have access to DE object but have its id, and it was easier to fetch object using this ID in handler than in place where the event is triggered.